rest-client/src/main/java/io/apicurio/registry/client/request/RequestHandler.java

```diff
@@ -59,5 +64,21 @@ private void handleError(Throwable e) {
             }
             logger.log(Level.SEVERE, "Error getting call result", e);
         }
+
+
+        private static void checkIfDeprecated(okhttp3.Headers headers) {
+            String isDeprecated = headers.get(Headers.DEPRECATED);
+            if (isDeprecated != null) {
```

Comment:
It's probably good enough to check for the presence of the header, but just to be sure you check to make sure the value is "true"?  The `HeadersHack` class sets the header value to `true`.